### DISCUSSION TITLE
Create widget with list of habits (similar to the main screen)

### DISCUSSION THREAD

**Discussion by ghost** [iSoron/uhabits#531] [Feature Requests]
Hi habit lovers! Having a simple condensed widget with habit list would take this app to another level. 
Example in picture. The idea is that you could quickly view and check habits directly from the phone desktop instead of opening the app. 

If anyone is capable of doing it I would be super grateful since I tried doing it myself but have no idea how to make it happen :(

![habits](https://user-images.githubusercontent.com/23101195/71088418-a8470f00-219e-11ea-9203-4644a6e1202c.png)

**Comment by iSoron** [comment]
Thank you for the suggestion, @SuperHeroWebDev. I agree that this feature could be very useful, but unfortunately I am not sure it's technically feasible. Android widgets are very restricted, and even our current (very simple) widgets already required multiple implementation hacks to overcome a number of non-obvious limitations. This would push it to another level.

If anyone is willing to try to code a prototype, that would be great, but I would expect it to be very hard. To clarify, rendering the widget could certainly be done, but accepting input and implementing scrolling would be very tricky, if not impossible.

**Reply by hiqua** [reply]
https://github.com/mpcjanssen/simpletask-android has scrolling in widgets, for reference. But the interactions with the list are really simple, it would be harder here.

**Comment by ghost** [comment]
Thanks for taking the time to clarify and answer. It's to bad the implementation would be so hard to do. Hopefully android widget development gets easier in the coming years. 

Thanks again for everything. For me still the best habits app out there so thank you on that.

**Comment by chennemann** [comment]
what we might be able to do is a not quite as sophisticated version. 
I'm not sure whether or not it is possible but we could maybe implement a widget that is not scrollable and that tries to fit a configurable amount of habits into the given space. by scaling the space and font size down accordingly.

**Comment by nyshocka** [comment]
I love this idea. FYI Nine has a scrolling setup for their Calendar Widget. I dont know the complexity though.

**Comment by chrrris1987** [comment]
I love this suggestion to see an overview in the widget, even though you could not checkmark it in the widget and you still get directed to the checking screen when you click it. I would make the widget size adjustable so you can make it more wide to see a whole week and also adjustable which habits you show. A way of seeing how you are doing overall with all the habits. I don't have any clue how to build this or if this is hard.. just saying what I would like to see as user :)

I would love if this would be possible for more (or all) widgets, like the ones that give a graphic dot or line how you are doing on a habit over time. Now I have 6 small ones, but how cool would it be to have one big one or if you split up a goal in smaller ones to see this group together. That you could choose which habits should be included in the graphic line/dots widget instead of picking one.

**Comment by chrrris1987** [comment]
![2020-02-03 08 52 05 1](https://user-images.githubusercontent.com/60602902/73685861-2e73d200-46c7-11ea-89fc-bb27e941b895.jpg)

For example like this. Above a widget which you maybe can mark, put portrays how you are doing these days. In the example you see all habits, but could be for just the food related habits. And if more widget can combine more habits, you could for example show a graphic line how you are doing

**Comment by Aquabrick** [comment]
I really support this idea. Im new to software development so had no idea this would be so tricky.
Given its usefulness, maybe a non-scrolling widget which simply shows the remaining un-ticked tasks for the day? Clicking would just open the app?

**Comment by amerker** [comment]
I also love this idea of having an overview of multiple habits. Something simple, non-scrolling would already be enough imo.

Specifically I'm thinking of a widget that just shows all active habits, with a (toggleable) checkmark for each habit, which corresponds to its state for the current day.
Which would be the same as creating a Checkmark widget for each habit individually (if you have many, this can get tiresome though, hence this idea).

**Comment by StefanMich** [comment]
I had a similar idea to this that I want to start investigating. My idea is to create an "Agenda" widget, that show the habits that are due today. That would help me focus on the habits that I actually need to. I think that will we compatible with the request in the OP.

I will start experimenting and get back when I have something working
